File: nexus/backend/crates/nexus-spi/src/dto/insight/get.rs
//! `GET /api/v1/insights` — a stored insight in the tenant.

use serde::{Deserialize, Serialize};
use serde_json::Value;
use utoipa::ToSchema;
use uuid::Uuid;

/// One stored insight: immutable id, name, the script, and its optional params
/// schema. The flat list carries every field the client needs to run or edit it.
#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ToSchema)]
pub struct InsightSummary {
    pub id: Uuid,
    pub name: String,
    /// The Rhai transform script.
    pub script: String,
    /// Optional JSON-Schema for the script's params.
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub params_schema: Option<Value>,
}

File: nexus/ui/src/api/insights/get.ts
import { fetchJson } from "@nube/starter-client-ts";
import type { StarterClient } from "@nube/starter-client-ts";

import type { InsightSummary } from "@/api/types";

// `GET /api/v1/insights/{id}` — full detail for one insight, including its
// Rhai transform script.
export function getInsight(
  client: StarterClient,
  id: string,
): Promise<InsightSummary> {
  return fetchJson<InsightSummary>(
    client,
    `${client.apiPrefix}/insights/${encodeURIComponent(id)}`,
  );
}
